javaslang/src/main/java/javaslang/collection/Collections.java

```diff
@@ -114,11 +119,11 @@ static int hash(Iterable<?> iterable) {
         return finisher.apply(cumulativeResult);
     }
 
-    static <T, U, C extends Iterable<U>, R extends Traversable<U>> R scanRight(
-            Iterable<? extends T> elements,
-            U zero, BiFunction<? super T, ? super U, ? extends U> operation,
-            C cumulativeResult, BiFunction<C, U, C> combiner, Function<C, R> finisher) {
-        final Iterator<? extends T> reversedElements = seq(elements).reverseIterator();
+    static <T, U, C extends Iterable<U>, R extends Traversable<U>> R scanRight(Iterable<? extends T> elements,
+                                                                               U zero, BiFunction<? super T, ? super U, ? extends U> operation,
+                                                                               C cumulativeResult, BiFunction<C, U, C> combiner,
+                                                                               Function<C, R> finisher) {
+        final Iterator<? extends T> reversedElements = seq(elements).reverseIterator(); // TODO a List will be reversed too many times this way
```

Comment:
We should add no TODOs in code (I know, I've also done it before). Better create an issue.